### PR TITLE
Avoid one source of sockets in CLOSE_WAIT state

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

**Comment by nono** [comment]
Travis still runs the build with js files but doesn't compile them itself...

**Comment by frankrousseau** [comment]
@nono Your PR breaks the tests.

File: server/controllers/applications.coffee [hunk 1]
```diff
@@ -12,41 +12,36 @@ getPathForStaticApp = (appName, path, root, callback) ->
...
 
+module.exports.app = (req, res, next) ->
+    req.url = req.url.substring "/apps/#{appName}".length
+    errTemplate = (err) ->
+        name: if err.code is 404 then 'not_found' else 'error_app'
+    forwardRequest req, res, errTemplate, next
+
 module.exports.publicApp = (req, res, next) ->
-    appName = req.params.name
     req.url = req.url.substring "/public/#{appName}".length
```
Comment (aenario) [inline server/controllers/applications.coffee]: you remove appName but it's still used here

File: server/controllers/applications.coffee [hunk 2]
```diff
@@ -12,41 +12,36 @@ getPathForStaticApp = (appName, path, root, callback) ->
...
-                name: 'error_public'
-            next error
-        else if result.type is 'static'
-            # showing public static app
-            getPathForStaticApp appName, req.url, result.path, (url) ->
-                send(req, url).pipe res
-        else
-            getProxy().web req, res, target: "http://localhost:#{result.port}"
+    errTemplate = (err) ->
+        name: 'error_public'
```
Comment (aenario) [inline server/controllers/applications.coffee]: are you not supposed to call forwardRequest here ?

File: server/controllers/applications.coffee (hunk 1 above)
Comment (nono) [inline server/controllers/applications.coffee]: @aenario thanks for doing my debug

File: server/controllers/devices.coffee [hunk 3]
```diff
@@ -61,7 +61,7 @@ deviceExists = (login, cb) ->
         else if body.length is 0
             cb null, false
         else
-            cb null, body[0]
+            cb null, body[0]?.value
```
Comment (nono) [inline server/controllers/devices.coffee]: @aenario I have no idea what have changed that causes this. Does it ring a bell for you?

Comment (aenario) [inline server/controllers/devices.coffee]: No idea, I checked the DS & Cradle too and couldn't find anything. 
Was this caught by travis ?

Comment (nono) [inline server/controllers/devices.coffee]: Yes, for example here: https://travis-ci.org/cozy/cozy-proxy/jobs/104924355#L753

Comment (aenario) [inline server/controllers/devices.coffee]: I think it is one if these case where things just happened to work : 

body[0] would be `{id: "xxx", key:"xxx", value:{actual doc}}`, we return this in callback as device.
It is truthy, so the `if device` in `checkLogin` will pass, then we do `device.id` in removeDevice, which pass too.

So this is a good change, but i dont think it was this one which fixed the error in travis :)

Comment (nono) [inline server/controllers/devices.coffee]: It is the source of the bug on travis. `checkLogin` is used for getting the actual doc when updating. Without it, the updated doc lost its `login` field. And when we try to delete the device, it is no longer found, as he has no login.

Comment (aenario) [inline server/controllers/devices.coffee]: Well then, One more "I have no idea how this ever worked"

**Pull request merged**